src/main.c

```diff
@@ -3583,25 +3586,26 @@ int sqlite3_open(
   sqlite3 **ppDb 
 ){
   return openDatabase(zFilename, ppDb,
-                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, NULL);
+                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, NULL, NULL);
 }
 int sqlite3_open_v2(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs        /* Name of VFS module to use */
 ){
-  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, NULL);
+  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, NULL, NULL);
 }
 
 int libsql_open(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs,       /* Name of VFS module to use, NULL for default */
-  const char *zWal        /* Name of WAL module to use */
+  const char *zWal,       /* Name of WAL module to use */
+  void* pMethodsData
 ) {
-  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, zWal);
+  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, zWal, pMethodsData);
```

Comment:
s/pMethodsData/pWalMethodsData -- here and in all other occurrences, where applicable